packages/caliper-fabric/lib/connector-versions/peer-gateway/PeerGateway.js

```diff
@@ -0,0 +1,442 @@
+            invokeStatus.SetStatusFail();
+            invokeStatus.SetVerification(true);
+            invokeStatus.SetResult('');
+            return invokeStatus;
+        }
+
+    }
+
+    /**
+     * Get the the specific contract for the identity invoking the request
+     * @param {string} mspId the mspId of the organisation that owns the identity
+     * @param {string} identityName the identity requested to be used by the workload
+     * @param {string} channelName the channel name the contract exists on
+     * @param {string} contractId the name of the contract to return
+     * @returns {Promise<FabricNetworkAPI.Contract>} A contract that may be used to submit or evaluate transactions
+     * @async
+     */
+    async _getContractForIdentityOnChannelWithChaincodeID(mspId, identityName, channelName, contractId) {
+        logger.debug('Entering _getContractForIdentityOnChannelWithChaincodeID');
+
+        const aliasName = this.connectorConfiguration.getAliasNameForOrganizationAndIdentityName(mspId, identityName);
+        const contractSet = this.contractInstancesByIdentity.get(aliasName);
+
+        // If no contract set found, there is a user configuration/test specification error, so it should terminate
+        if (!contractSet) {
+            throw Error(`No contracts for invokerIdentity ${identityName}${mspId ? ` in ${mspId}` : ''} found. Identity and/or MspId does not exist`);
+        }
+
+        const contract = contractSet.get(`${channelName}_${contractId}`);
+
+        // If no contract found, there is a user configuration/test specification error, so it should terminate
+        if (!contract) {
+            throw Error(`Unable to find specified contract ${contractId} on channel ${channelName}!`);
+        }
+
+        logger.debug('Exiting _getContractForIdentityOnChannelWithChaincodeID');
+
+        return contract;
+    }
+
+
+    /**
+     * Create a grpc/grpcs client associated with a specific peer
+     * @param {string} mspId The msp id of the organisation which owns the identity
+     * @param {string} peer The peer we are creating a grpc client for
+     * @returns {Promise<GrpcClient>} a grpc client for the passed peer
+     * @async
+     */
+    async _createClientForPeer(mspId, peer){
+        let client;
+        if (this.clients.get(peer)) { //check if client is aready created for specified peer
+            client = this.clients.get(peer);
+        } else {
+            const connectionProfileDefinition = await this.connectorConfiguration.getConnectionProfileDefinitionForOrganization(mspId);
+
+            //set grpc options
+            const grpcOptions = connectionProfileDefinition.getGrpcOptionsForPeer(peer);
+            //extrapolate and set grpc option hostnameOverride from the connection profile
+            if (grpcOptions.hasOwnProperty('hostnameOverride')){
```

Comment:
hostnameOverride I believe is only relevant for the Java SDK which is why its put into the connection profile, however we can just ignore it which is why it's not in the above list of grpc options and it doesn't matter if it's present in the connection profile it will be ignored.